content/2023-05-03-beware-of-unified-builds.rst

```diff
@@ -0,0 +1,317 @@
+How Jumbo builds lurked into the Firefox Build System
+#####################################################
+
+:date: 2023-05-03
+:category: mozilla
+:lang: en
+:authors: Serge Guelton
+:summary: Jumbo builds make C++ code compilation faster, but what happens when
+          you require them?
+
+
+Jumbo Build
+===========
+
+Jumbo builds, a.k.a. unified build or unity builds, is a technique that consists in
+concatenating several C or C++ files in one before compiling. This generally
+leads to faster compilation time because it aggregates the cost of parsing the
+same headers over and over.
+
+It is one of the many approach one can use to reduce C++ software compilation
+time, alongside precompiled headers and C++20 modules.
+
+It is not an obscure technique: it's officially supported by CMake through
+``CMAKE_UNITY_BUILD`` (see
+https://cmake.org/cmake/help/latest/variable/CMAKE_UNITY_BUILD.html)
+
+At some point, chromium has been using it:
+https://chromium.googlesource.com/chromium/src.git/+/65.0.3283.0/docs/jumbo.md,
+even if they get rid of it afterward
+https://groups.google.com/a/chromium.org/g/chromium-dev/c/DP9TQszzQLI
+
+It is also supported by the internal build system used at Mozilla. And the
```

Comment:
"at Mozilla" isn't very precise, we do a lot more than Firefox.

> used at Mozilla for the Firefox code base.

`s/And //`.